src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -423,17 +423,33 @@ func (gc *GarbageCollector) removeUntaggedBlobs(ctx job.Context) {
 			continue
 		}
 		p := result.Data
-		all, err := gc.blobMgr.List(ctx.SystemContext(), blob.ListParams{
-			ProjectID:  p.ProjectID,
-			UpdateTime: time.Now().Add(-time.Duration(gc.timeWindowHours) * time.Hour),
-		})
-		if err != nil {
-			gc.logger.Errorf("failed to get blobs of project, %v", err)
-			continue
-		}
-		if err := gc.blobMgr.CleanupAssociationsForProject(ctx.SystemContext(), p.ProjectID, all); err != nil {
-			gc.logger.Errorf("failed to clean untagged blobs of project, %v", err)
-			continue
+
+		pn := int64(1)
+		for {
+			rg := q.Range{
+				Max: time.Now().Add(-time.Duration(gc.timeWindowHours) * time.Hour).Format(time.RFC3339),
```

Comment:
Should we move this line out of the for loop?  Otherwise, the max value will be changed in every loop, this may affect the stability of the results.

Reply:
will do, but put it as here will not impace the results.